transport/server.cc

```diff
@@ -1952,6 +1953,11 @@ class type_codec {
             }
             return;
         }
+        if (type->is_vector()) {
+            r.write_short(uint16_t(type_id::CUSTOM));
+            r.write_string(type->name());
```

Comment:
I'm surprised that they didn't add a new type id. Maybe it's a good precedent for us: In the future, we can add more types that we can invent as a Scylla extension and encode them in the transport in exactly this way, and we won't need to worry about breaking Cassandra compatibility (since we can't just pick type 0x22, in fear one day Cassandra will start to use it).

Reply:
It's not surprising, the type was first added to Datastax' fork (which couldn't claim a new type ID), then was contributed to Cassandra without changing the type ID because it would break drivers.

Reply:
But as you can see, this never happened before. I'm pretty sure the Datastax guys could have coordinated with Apache guys (same guys? :-)) to pick a number. Anyway, as I said, it's a good precedent in case in the future we'll want to add our own types. Too bad other things other aspects of the protocol don't have this CUSTOM trick - for example once I had the idea to add a new consistency level, but there is no way to do it without picking a number and hoping Cassandra don't pick the same number in the future.